include/array/array.h

```diff
@@ -2870,6 +2871,13 @@ const_array_ref<U, Shape> reinterpret(const array<T, Shape, Alloc>& a) {
   return reinterpret<const U>(a.cref());
 }
 
+/** Reinterpret a const_array_ref as a array_ref. This "reinterprets away the
```

Comment:
I think the comment is a bit stale from older versions maybe?

Reply:
How about:

```
/** Reinterpret the const_array_ref `a` of type `T` (aka array_ref<const T>) to have a different type
 * `U` using `const_cast`.
 */
```

Reply:
LGTM